gammapy/spectrum/dataset.py

```diff
@@ -359,6 +359,38 @@ def plot_residuals(self, method="diff", ax=None, **kwargs):
         ax.set_ylabel("Residuals ({})".format(label))
         return ax
 
+    @classmethod
+    def create(cls, e_reco, e_true=None, reference_time="2000-01-01"):
+        """Creates empty SpectrumDataset
+
+        Empty containers are created with the correct geometry.
+        counts, background and aeff are zero and edisp is diagonal.
+
+        Parameters
+        ----------
+        e_reco : `~astropy.units.Quantity`
+            edges of counts vector
+        e_true : `~astropy.units.Quantity`
+            edges of effective area table. If not set use reco energy values. Default : None
+        reference_time : `~astropy.time.Time`
+            reference time of the dataset, Default is "2000-01-01"
+        """
+        counts = CountsSpectrum(e_reco[:-1], e_reco[1:])
+        background = CountsSpectrum(e_reco[:-1], e_reco[1:])
+        aeff = EffectiveAreaTable(e_true[:-1],e_true[1:], np.zeros(e_true[:-1].shape)*u.m**2)
+        edisp = EnergyDispersion.from_diagonal_response(e_true, e_reco)
+        mask_safe = np.ones_like(counts.data.data, 'bool')
```

Comment:
Note that `CountsSpectrum.data` is now a Numpy array and not an `NDDataArray`anymore, so writing `counts.data` is sufficient.

Reply:
Right, corrected.